pkg/config/config.go

```diff
@@ -1,26 +1,102 @@
 package config
 
+import (
+	"fmt"
+
+	"github.com/caarlos0/env"
+	"github.com/golang/glog"
+)
+
 type DaemonConfig struct {
-	Plugin         string `json:"plugin"`          // Subnet manager plugin name
-	PeriodicUpdate int    `json:"periodic_update"` // Interval between every check for the added and deleted pods
+	PeriodicUpdate int `env:"PERIODIC_UPDATE"` // Interval between every check for the added and deleted pods
 	GuidPool       GuidPoolConfig
 	SubnetManager  SubnetManagerPluginConfig
 }
 
 type GuidPoolConfig struct {
-	GuidRangeStart string `json:"guid_range_start"` // First guid of the pool
-	GuidRangeEnd   string `json:"guid_range_end"`   // Last of the guid pool
+	RangeStart string `env:"RANGE_START"` // First guid of the pool
```

Comment:
the env variable name is abit cryptic on its own, perhaps GUID_POOL_RANGE_START, GUID_POOL_RANGE_END
i think before it was prefixed by GUID IIRC 

A general guideline could be to name the environment variables like : [component name_]<attribute name> for global daemon attributes we would have the attribute name flat e.g `PERIODIC_UPDATE` (or we could prefix it by "DAEMON" e.g `DAEMON_PERIODIC_UPDATE`)

Reply:
and for subnet manager config we can prefix with `SM_`